### PR TITLE
[Reply] Contrast aware updates.

### PR DESCRIPTION
**Pull request by jwill**
Added colors for standard, medium, and high in both light and dark themes. 
Added helper function to select contrast level based on values from SysUI controls. 
Added more variance in application of surface container roles to better indicate elevation and emphasis.

### CODE REVIEW

File: Reply/app/src/main/java/com/example/reply/ui/components/ReplyEmailThreadItem.kt [hunk 1]
```diff
@@ -107,31 +111,32 @@ fun ReplyEmailThreadItem(
...
+                        containerColor = MaterialTheme.colorScheme.surfaceBright
                     )
                 ) {
                     Text(
                         text = stringResource(id = R.string.reply),
-                        color = MaterialTheme.colorScheme.onSurfaceVariant
+                        color = MaterialTheme.colorScheme.onSurface
                     )
                 }
+                Spacer(modifier = Modifier.width(8.dp))
```
Comment (IanGClifton) [inline Reply/app/src/main/java/com/example/reply/ui/components/ReplyEmailThreadItem.kt:129]: Nit: This spacing is a combination of line 115 and line 129. I'd combine them to just do 12dp in one place, probably 115, but I don't have a strong preference whether it's that or a spacer with no horizontalArrangement

File: Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt [hunk 2]
```diff
@@ -15,95 +15,332 @@
...
+            window.statusBarColor = replyColorScheme.primary.toArgb()
+            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
+        }
     }
 
     MaterialTheme(
         colorScheme = replyColorScheme,
-        typography = replyTypography,
-        shapes = shapes,
+        typography = replyTypography, shapes = shapes,
```
Comment (IanGClifton) [inline Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt:343]: Nit: Should still be separate lines

File: Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt [hunk 3]
```diff
@@ -15,95 +15,332 @@
...
+
+    MaterialTheme(
+        colorScheme = replyColorScheme,
+        typography = replyTypography, shapes = shapes,
+        content = content
+    )
+}
+
 @Composable
 fun ReplyTheme(
```
Comment (IanGClifton) [inline Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt:317]: Is the ReplyTheme still used anywhere?

**Review by IanGClifton — COMMENTED** [review]
Very minor feedback; I'm assuming all the colors are perfect and beautiful in every way

File: Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt (hunk 3 above)
Comment (jwill) [reply Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt:317]: It was being used in previews but I've replaced all those references and removed ReplyTheme. Done.

File: Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt (hunk 2 above)
Comment (jwill) [reply Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt:343]: Done.

File: Reply/app/src/main/java/com/example/reply/ui/components/ReplyEmailThreadItem.kt (hunk 1 above)
Comment (jwill) [reply Reply/app/src/main/java/com/example/reply/ui/components/ReplyEmailThreadItem.kt:129]: Done.

**Review by IanGClifton — APPROVED** [review]

**Pull request merged**